assets/src/hooks/v2/use_interval_paging.tsx

```diff
@@ -0,0 +1,57 @@
+import { useState, useEffect, useRef } from "react";
+
+interface UseIntervalPagingProps {
+  numPages: number; // Total number of pages to cycle through
+  cycleIntervalMs: number; // In milliseconds, the interval for cycling pagesß
+  onFinish: () => void; // Callback when cycling completes
+}
+
+/**
+ * This hook acts as a way for components with multiple pages to advance
+ * through their pages on a set time interval.
+ */
+function useIntervalPaging({
+  numPages,
+  cycleIntervalMs,
+  onFinish,
+}: UseIntervalPagingProps) {
+  const [pageIndex, setPageIndex] = useState(0);
+  const [isFinished, setIsFinished] = useState(false);
+
+  // Use ref to keep stable references to timer interval without triggering re-renders
+  const intervalRef = useRef<NodeJS.Timeout | null>(null);
+
+  useEffect(() => {
+    intervalRef.current = setInterval(() => {
+      setPageIndex((prevPage) => {
+        const nextPage = (prevPage + 1) % numPages;
+
+        if (nextPage === 0) {
+          // Updates the state of the data being displayed in case of any refreshes before cycling the page to index 0.
+          setIsFinished(true);
+        }
+
+        return nextPage;
+      });
+    }, cycleIntervalMs);
+
+    // Cleanup interval on unmount
+    return () => {
+      if (intervalRef.current) {
+        clearInterval(intervalRef.current);
+      }
+    };
+  }, [cycleIntervalMs, numPages]);
+
+  // Handles calling the onFinish function in persistentWrapper.
+  // This needs to be done in a separate hook outside of the setInterval callback.
```

Comment:
Why is this? Similar to the comment about `isFinished` in the persistent wrapper, it seems like we should be able to handle this by directly calling functions rather than setting internal bits of state that only exist to trigger effects.

Reply:
The problem is that you can't update the state of persistent wrapper within the callback function of `setInterval`. This is a weird piece of React I wasn't familiar with. I got around this by having this boolean flag `isFinished` that serves only to trigger this hook.

Reply:
I updated the naming and comment here for more clarity.